compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -45,6 +45,21 @@ pub struct ModuleItems {
     foreign_items: Box<[ForeignItemId]>,
 }
 
+impl ModuleItems {
+    pub fn items(&self) -> impl Iterator<Item = ItemId> {
+        self.items.to_vec().into_iter()
```

Comment:
`to_vec` allocates needlessly. Likewise in the other methods.
```suggestion
        self.items.iter().copied()
```